supernode/daemon/mgr/gc/gc_manager.go

```diff
@@ -20,14 +20,40 @@ import (
 	"context"
 	"time"
 
+	"github.com/dragonflyoss/Dragonfly/pkg/metricsutils"
 	"github.com/dragonflyoss/Dragonfly/supernode/config"
 	"github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr"
 
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/sirupsen/logrus"
 )
 
 var _ mgr.GCMgr = &Manager{}
 
+type metrics struct {
+	gcTaskCount    *prometheus.CounterVec
+	gcTaskDuration *prometheus.HistogramVec
+	gcPeerCount    *prometheus.CounterVec
+	gcPeerDuration *prometheus.HistogramVec
+}
+
+func newMetrics(register prometheus.Registerer) *metrics {
+	return &metrics{
+		gcTaskCount: metricsutils.NewCounter(config.SubsystemSupernode, "gc_task_total",
+			"Total times of task gc", []string{}, register),
+		gcTaskDuration: metricsutils.NewHistogram(config.SubsystemSupernode, "gc_task_duration_seconds",
+			"Histogram of latencies for task gc", []string{"task"},
+			[]float64{.1, .2, .4, 1, 3, 8, 20, 60, 120}, register,
```

Comment:
Actually I am curious about the buckets you select. Why do you think the  gc duration can reach 20s,60s even 120s? I think it is not possible and meaningless, only increases memory usage.  I think use milliseconds or other smaller units is better.

Reply:
Also I would say the label "task" will cause cardinality problem, please remove it. We don't recommend to use it.